<a href="https://colab.research.google.com/github/steinruck/WANG-690-FALL-2020/blob/master/Assignment_10/Assignment_10.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Assignment # 10 - Point Estimate and Interval Estimate (Confidence Interval)
A random survey of enrollment at 35 community colleges across the United States yielded the following figures:

6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;

5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200;

17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861;

1,263; 7,285; 28,165; 5,080; 11,622

Perform point estimate and interval estimate with 95% confidence level using t-distribution.

Since we don't know the population variance, we use t-distribiution instead of normal distribution.

In [14]:
# Import Python packages
import math
import pandas as pd
import numpy as np
import scipy.stats as stats
import matplotlib.pyplot as plt


### Step 0 - Data Preprocessing

Process the raw data to make a list of integers. In order to calculate descriptive statistic, Python needs to work with a list of numbers.

In [1]:
# make each line of numbers a string object and then concatenate them together 
# The end result is one single string containing 35 numbers separated by ";"

line1 = "6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;"
line2 = "5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200;"
line3 = "17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861;"
line4 = "1,263; 7,285; 28,165; 5,080; 11,622"

full = line1 + line2 + line3 + line4
print(full)

6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200;17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861;1,263; 7,285; 28,165; 5,080; 11,622


In [2]:
x = full.split(";")
print(x)

['6,414', ' 1,550', ' 2,109', ' 9,350', ' 21,828', ' 4,300', ' 5,944', ' 5,722', ' 2,825', ' 2,044', '5,481', ' 5,200', ' 5,853', ' 2,750', ' 10,012', ' 6,357', ' 27,000', ' 9,414', ' 7,681', ' 3,200', '17,500', ' 9,200', ' 7,380', ' 18,314', ' 6,557', ' 13,713', ' 17,768', ' 7,493', ' 2,771', ' 2,861', '1,263', ' 7,285', ' 28,165', ' 5,080', ' 11,622']


In [3]:
clean = []
for i in x:
    y = i.replace(",", "")
    y = int(y)
    clean.append(y)
    
print(clean)

[6414, 1550, 2109, 9350, 21828, 4300, 5944, 5722, 2825, 2044, 5481, 5200, 5853, 2750, 10012, 6357, 27000, 9414, 7681, 3200, 17500, 9200, 7380, 18314, 6557, 13713, 17768, 7493, 2771, 2861, 1263, 7285, 28165, 5080, 11622]


### Step 1 - Calculate and Display the Sample Size and Sample Mean

In [5]:
# Calculate and display the sample size

sample_size = len(clean)
print("Sample Size =", sample_size)

Sample Size = 35


In [9]:
# Calculate and display the sample mean

sample_mean = round(np.mean(clean))
print("Sample Mean =", sample_mean)

Sample Mean = 8629.0


The point estimate of the mean enrollment of US community colleges is 8629.


### Step 2 - Calculate and Display the Sample Standard Deviation & Sample Standard Error

Sample Standard Deviation $S=\sqrt{\dfrac{1}{n-1}\sum\limits_{i=1}^n (X_i-\bar{X})^2}$

Sample Standard Error = $\dfrac{S}{\sqrt{n}}$

Note: The default Delta Degree of Freedom (DDOF) for Numpy's std function is 0 which is applicable to populate data. For sample data, we need to specify ddof=1.

For the enrollment data, we round up the statistics to be the full integers (no decimal points).

In [11]:
# Calculate and display the sample standard deviation using Numpy's std function.

standard_dev = np.std(clean, ddof=1) 
standard_dev = round(standard_dev)         
print("Sample Standard Deviation =", standard_dev)

Sample Standard Deviation = 6944.0


In [12]:
# Calculate and display the sample standard error

standard_err = standard_dev / math.sqrt(sample_size)
standard_err = round(standard_err)
print("Sample Standard Error is", standard_err)

Sample Standard Error is 1174.0



### Step 3 - Calculate t Critical Value using t-Distribution

$\alpha$ = 1 - Confidence Level = 1 - 95% = 0.05

$\dfrac{\alpha}{2}$ = 0.025

n (sample size) = 35

df (degree of freedom) = n - 1 = 35 - 1 = 34

We will use Python scipy.stats t-distribution's PPF (Percentage Point Function) to calculate t critical value $t_{0.025,34}$.

In [27]:
# Calculate and display the t critical value using scipy.stats.t package ppf function

t_value = stats.t.ppf(0.975, 34)
print("t critical value = ", round(t_value,2))

t critical value =  2.03


### Step 4 - Calculate the Margin of Error

Margin of Error = t-Statistics * Sample Standard Error = $t_{\alpha/2,n-1}\left(\dfrac{s}{\sqrt{n}}\right)$

In [36]:
# Calculate and display the margin of error

margin_error = round(t_value * standard_err) 
print("Margin of Error =", margin_error)

Margin of Error = 2386.0


### Step 5 - Calculate Lower and Upper Limit of the Confidence Interval

Lower Limit = Sample Mean - Margin of Error

Upper Limit = Sample Mean + Margin of Error

In [37]:
# Calculate and display the lower limit

lower_limit = sample_mean - margin_error
print("Lower Limit =", lower_limit)

Lower Limit = 6243.0


In [30]:
# Calculate and display the upper limit
upper_limit = sample_mean + margin_error
print("Upper Limit =", upper_limit)

Upper Limit = 11015.0


### Step 6 - Now We have the 95% Confidence Interval

Confidence Interval ($\sigma$ unknown) = $\bar{x}\space\pm\space t_{\alpha/2}\left(\dfrac{S}{\sqrt{n}}\right)$ = Sample_Mean $\pm$ Margin of Error

In [32]:
# Print the 95% confidence interveral in the form (lower limit, upper limit)
print("The 95% Confidence Interval = (", lower_limit, ",", upper_limit, ")")

The 95% Confidence Interval = ( 6243.0 , 11015.0 )
